[ci skip] Update changelog

ndellingwood opened the pull request.

masterleinad commented on CHANGELOG.md:
```diff
@@ -1,5 +1,104 @@
+
+#### OpenACC:
+* Add atomics support [\#6446](https://github.com/kokkos/kokkos/pull/6446)
+* Change the default execution policy behavior of the OpenACC backend [\#6772](https://github.com/kokkos/kokkos/pull/6772)
```
```suggestion
* Make the OpenACC backend asynchronous [\#6772](https://github.com/kokkos/kokkos/pull/6772)
```

masterleinad commented on CHANGELOG.md:
```diff
@@ -1,5 +1,104 @@
+
+#### OpenMP:
+* Improve performance of view initializations and filling with zeros [\#6573](https://github.com/kokkos/kokkos/pull/6573)
+* Use omp_get_max_active_levels() when supported [\#6666]((https://github.com/kokkos/kokkos/pull/6666) 
```
```suggestion
```

masterleinad commented on CHANGELOG.md:
```diff
@@ -1,5 +1,104 @@
+* CTAD (deduction guides) for MDRangePolicy [\#5516]
+
+### Build System Changes 
+* Atomics always enabled by default (bug fix?).  Add `ATOMICS_BYPASS` configuration option to disable atomics [\#6692](https://github.com/kokkos/kokkos/pull/6692)
```
```suggestion
* Atomics always enabled by default [\#6692](https://github.com/kokkos/kokkos/pull/6692)
```
Yeah, more like a bug fix.

ndellingwood commented:
Cross referencing ##6519

dalg24 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* Drop support for old CPU architectures (`ARCH_BGQ`, `ARCH_POWER7`, `ARCH_WSM` and associated `ARCH_SSE4` macro) [\#6806](https://github.com/kokkos/kokkos/pull/6806)
+
+### Bug Fixes
+* Fix Team level MDRange reductions: [\#6511](https://github.com/kokkos/kokkos/pull/6511)
```
```suggestion
* Fix team-level MDRange reductions: [\#6511](https://github.com/kokkos/kokkos/pull/6511)
```

dalg24 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+
+### Bug Fixes
+* Fix Team level MDRange reductions: [\#6511](https://github.com/kokkos/kokkos/pull/6511)
+* Fix CUDA and SYCL small value type (16bit) team reductions [\#5334](https://github.com/kokkos/kokkos/pull/5334)
```
```suggestion
* Fix CUDA and SYCL small value type (16-bit) team reductions [\#5334](https://github.com/kokkos/kokkos/pull/5334)
```

dalg24 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* Drop support for deprecated command-line arguments and environment variables [\#6744](https://github.com/kokkos/kokkos/pull/6744)
+* Deprecate `ExecutionSpace::in_parallel()` [\#6582](https://github.com/kokkos/kokkos/pull/6582)
+* Remove `Experimental::HBWSpace` and support for linking against memkind [\#6791](https://github.com/kokkos/kokkos/pull/6791)
+* Drop `KOKKOS_ENABLE_INTEL_MM_ALLOC` macro [#6797](https://github.com/kokkos/kokkos/pull/6797)
```
Should that one go to "incompatibilities"

ndellingwood replied:
Fixed in [9200491](https://github.com/kokkos/kokkos/pull/6886/commits/9200491aab44d1d187673c44d9435ebcc3b02b6c)

ndellingwood commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* Enable `{transform_}exclusive_scan` in place [\#6667](https://github.com/kokkos/kokkos/pull/6667)
+* `fill_random` overload that do not take an execution space instance argument should fence [\#6658](https://github.com/kokkos/kokkos/pull/6658)
+* HIP,Cuda,OpenMPTarget: Fixup use provided execution space when copying host inaccessible reduction result [\#6777](https://github.com/kokkos/kokkos/pull/6777)
+* Fix typo in `cuda_func_set_attribute[s]_wrapper` [\#6786](https://github.com/kokkos/kokkos/pull/6786)
```
Fixed in [9200491](https://github.com/kokkos/kokkos/pull/6886/commits/9200491aab44d1d187673c44d9435ebcc3b02b6c)

ndellingwood commented:
@dalg24 I addressed your remaining comments in [9200491](https://github.com/kokkos/kokkos/pull/6886/commits/9200491aab44d1d187673c44d9435ebcc3b02b6c), fixed a couple more typos I found as well

ajpowelsnl commented on CHANGELOG.md:
```diff
@@ -1,5 +1,101 @@
+* Drop librt TPL and associated `KOKKOS_ENABLE_LIBRT` macro [\#6798](https://github.com/kokkos/kokkos/pull/6798)
+* Drop support for old CPU architectures (`ARCH_BGQ`, `ARCH_POWER7`, `ARCH_WSM` and associated `ARCH_SSE4` macro) [\#6806](https://github.com/kokkos/kokkos/pull/6806)
+
+### Bug Fixes
```
**TODO:**
- Cherrypick #6892 to release branch
- Add to CHANGELOG [6892](https://github.com/kokkos/kokkos/pull/6892):  Fix a bug in Makefile when using AMD GPU architectures

ndellingwood commented:
@dalg24 @masterleinad @ajpowelsnl do we anticipate any more updates to release-candidate-4.3.00? If not, this should be ready for merge I believe

masterleinad commented:
> @dalg24 @masterleinad @ajpowelsnl do we anticipate any more updates to release-candidate-4.3.00? If not, this should be ready for merge I believe

I'd still like to discuss if we want to act upon the issue https://github.com/kokkos/kokkos/pull/6898 is fixing in any way for the release.

ndellingwood commented:
I added an entry for #6898 as a fix under Build System, sha [6a16ef1](https://github.com/kokkos/kokkos/pull/6886/commits/6a16ef11d60787bf9727f0cf97530d640ce755e5)

I believe the cherry-pick of #6898 was the final planned item for the release, @dalg24 @crtrott @masterleinad shall we move forward with merge of this PR?

dalg24 commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* Add atomics support [\#6446](https://github.com/kokkos/kokkos/pull/6446)
+* Make the OpenACC backend asynchronous [\#6772](https://github.com/kokkos/kokkos/pull/6772)
+
+### Threads:
```
```suggestion
#### Threads:
```

masterleinad commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* `fill_random` overload that do not take an execution space instance argument should fence [\#6658](https://github.com/kokkos/kokkos/pull/6658)
+* HIP,Cuda,OpenMPTarget: Fixup use provided execution space when copying host inaccessible reduction result [\#6777](https://github.com/kokkos/kokkos/pull/6777)
+* Fix typo in `cuda_func_set_attribute[s]_wrapper` preventing proper setting of desired occupancy [\#6786](https://github.com/kokkos/kokkos/pull/6786)
+* Clean up shift_{right, left}_team_impl [\#6821](https://github.com/kokkos/kokkos/pull/6821)
```
```suggestion
* Avoid undefined behavior due to conversion between signed and unsigned integers in shift_{right, left}_team_impl [\#6821](https://github.com/kokkos/kokkos/pull/6821)
```

crtrott commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+ * When rocthrust is available use it when sorting [\#6793](https://github.com/kokkos/kokkos/pull/6793)
+
+#### SYCL:
+* Error out on initialization if the backend is different from `ext_oneapi_*` [\#6784](https://github.com/kokkos/kokkos/pull/6784)
+* Filter GPU devices for `ext_onapi_*` GPU devices [\#6758](https://github.com/kokkos/kokkos/pull/6784)
```
```suggestion
* We only support OneAPI SYCL implementation: add check during initialization
  * Error out on initialization if the backend is different from `ext_oneapi_*` [\#6784](https://github.com/kokkos/kokkos/pull/6784)
  * Filter GPU devices for `ext_onapi_*` GPU devices [\#6758](https://github.com/kokkos/kokkos/pull/6784)
```

crtrott commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+#### SYCL:
+* Error out on initialization if the backend is different from `ext_oneapi_*` [\#6784](https://github.com/kokkos/kokkos/pull/6784)
+* Filter GPU devices for `ext_onapi_*` GPU devices [\#6758](https://github.com/kokkos/kokkos/pull/6784)
+* Avoid unnecessary zero-memset of the scratch flags in SYCL [\#6739](https://github.com/kokkos/kokkos/pull/6739)
+* Address deprecations after oneAPI 2023.2.0 [\#6577](https://github.com/kokkos/kokkos/pull/6739)
+* Use host-pinned memory to copy reduction/scan result [\#6500](https://github.com/kokkos/kokkos/pull/6500)
```
```suggestion
* Performance Improvements
  * Avoid unnecessary zero-memset of the scratch flags in SYCL [\#6739](https://github.com/kokkos/kokkos/pull/6739)
  * Use host-pinned memory to copy reduction/scan result [\#6500](https://github.com/kokkos/kokkos/pull/6500)
* Address deprecations after oneAPI 2023.2.0 [\#6577](https://github.com/kokkos/kokkos/pull/6739)
```

crtrott commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+
+#### HIP:
+ * Fix compilation when using amdclang (with ROCm >= 5.7) and RDC [\#6857](https://github.com/kokkos/kokkos/pull/6857)
+ * When rocthrust is available use it when sorting [\#6793](https://github.com/kokkos/kokkos/pull/6793)
```
```suggestion
 * Use rocthrust for sorting, when available [\#6793](https://github.com/kokkos/kokkos/pull/6793)
```

crtrott commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* Fix/improvement to `remove_if` parallel algorithm: use the provided execution space instance for temporary allocations and drop unnecessaryinitialization + avoid evaluating twice the predicate during final pass [\#6747](https://github.com/kokkos/kokkos/pull/6747)
+* Add runtime function to query the number of devices and make device ID consistent with `KOKKOS_VISIBLE_DEVICES` [\#6713](https://github.com/kokkos/kokkos/pull/6713)
+* simd: support `vector_aligned_tag` [\#6243](https://github.com/kokkos/kokkos/pull/6243)
+* Prefer defaulted default constructor for Bitset [\#6524](https://github.com/kokkos/kokkos/pull/6524)
```
```suggestion
```

crtrott commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* Use LLVM extensions for dynamic shared memory [\#6380](https://github.com/kokkos/kokkos/pull/6380)
+* Guard scratch memory usage in ParallelReduce [\#6585 ](https://github.com/kokkos/kokkos/pull/6585)
+* Update linker flags for Intel GPUs update [\#6735](https://github.com/kokkos/kokkos/pull/6735)
+* Improve handling of printf in OMPT on Intel GPUs [\#6652](https://github.com/kokkos/kokkos/pull/6652)
```
```suggestion
* Improve handling of printf on Intel GPUs [\#6652](https://github.com/kokkos/kokkos/pull/6652)
```

crtrott commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* CTAD (deduction guides) for MDRangePolicy [\#5516](https://github.com/kokkos/kokkos/pull/5516)
+
+### Build System Changes
+* Atomics always enabled by default [\#6692](https://github.com/kokkos/kokkos/pull/6692)
```
```suggestion
* Require `Kokkos_ENABLE_ATOMICS_BYPASS` option to bypass atomic operation for Serial backend only builds [\#6692](https://github.com/kokkos/kokkos/pull/6692)
```

crtrott commented on CHANGELOG.md:
```diff
@@ -1,5 +1,103 @@
+* Tools(profiling): fix typo Kokkos_Tools_Optim[i]zationGoal [\#6642](https://github.com/kokkos/kokkos/pull/6642)
+* Remove variadic range policy constructor (disallow passing multiple trailing chunk size arguments) [\#6845](https://github.com/kokkos/kokkos/pull/6845)
+* Improve message on view out of bounds access and always abort [\#6861](https://github.com/kokkos/kokkos/pull/6861)
+* Drop `KOKKOS_ENABLE_INTEL_MM_ALLOC` macro [\#6797](https://github.com/kokkos/kokkos/pull/6797)
+
+### Deprecations
+* Remove `Kokkos::Experimental::LogicalMemorySpace` (without going through deprecation) [\#6557](https://github.com/kokkos/kokkos/pull/6557)
```
```suggestion
* Drop `KOKKOS_ENABLE_INTEL_MM_ALLOC` macro [\#6797](https://github.com/kokkos/kokkos/pull/6797)
* Remove `Kokkos::Experimental::LogicalMemorySpace` (without going through deprecation) [\#6557](https://github.com/kokkos/kokkos/pull/6557)

### Deprecations
```

crtrott commented on CHANGELOG.md:
```diff
@@ -1,5 +1,105 @@
+* Remove variadic range policy constructor (disallow passing multiple trailing chunk size arguments) [\#6845](https://github.com/kokkos/kokkos/pull/6845)
+* Improve message on view out of bounds access and always abort [\#6861](https://github.com/kokkos/kokkos/pull/6861)
+* Drop `KOKKOS_ENABLE_INTEL_MM_ALLOC` macro [\#6797](https://github.com/kokkos/kokkos/pull/6797)
+* Remove `Kokkos::Experimental::LogicalMemorySpace` (without going through deprecation) [\#6557](https://github.com/kokkos/kokkos/pull/6557)
+
+### Deprecations
+* Provide kokkos_swap as part of Core and deprecate Experimental::swap in Algorithms [\#6697](https://github.com/kokkos/kokkos/pull/6697)
+* Deprecate {Cuda,HIP}::detect_device_count() and Cuda::[detect_]device_arch() [\#6710](https://github.com/kokkos/kokkos/pull/6710)
+* Drop support for deprecated command-line arguments and environment variables [\#6744](https://github.com/kokkos/kokkos/pull/6744)
+* Deprecate `ExecutionSpace::in_parallel()` [\#6582](https://github.com/kokkos/kokkos/pull/6582)
+* Remove `Experimental::HBWSpace` and support for linking against memkind [\#6791](https://github.com/kokkos/kokkos/pull/6791)
+* Drop librt TPL and associated `KOKKOS_ENABLE_LIBRT` macro [\#6798](https://github.com/kokkos/kokkos/pull/6798)
+* Drop support for old CPU architectures (`ARCH_BGQ`, `ARCH_POWER7`, `ARCH_WSM` and associated `ARCH_SSE4` macro) [\#6806](https://github.com/kokkos/kokkos/pull/6806)
```
```suggestion
* Remove `Kokkos::Experimental::LogicalMemorySpace` (without going through deprecation) [\#6557](https://github.com/kokkos/kokkos/pull/6557)
* Remove `Experimental::HBWSpace` and support for linking against memkind [\#6791](https://github.com/kokkos/kokkos/pull/6791)
* Drop librt TPL and associated `KOKKOS_ENABLE_LIBRT` macro [\#6798](https://github.com/kokkos/kokkos/pull/6798)
* Drop support for old CPU architectures (`ARCH_BGQ`, `ARCH_POWER7`, `ARCH_WSM` and associated `ARCH_SSE4` macro) [\#6806](https://github.com/kokkos/kokkos/pull/6806)
* Drop support for deprecated command-line arguments and environment variables [\#6744](https://github.com/kokkos/kokkos/pull/6744)

### Deprecations
* Provide kokkos_swap as part of Core and deprecate Experimental::swap in Algorithms [\#6697](https://github.com/kokkos/kokkos/pull/6697)
* Deprecate {Cuda,HIP}::detect_device_count() and Cuda::[detect_]device_arch() [\#6710](https://github.com/kokkos/kokkos/pull/6710)
* Deprecate `ExecutionSpace::in_parallel()` [\#6582](https://github.com/kokkos/kokkos/pull/6582)
```
